x/stakeibc/keeper/hooks.go

```diff
@@ -265,7 +271,11 @@ func (k Keeper) ReinvestRewards(ctx sdk.Context) {
 		}
 		k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "BlockTime for host zone: %d", blockTime))
 
-		err = k.UpdateWithdrawalBalance(ctx, hostZone)
+		// Submit a withdrawal balance ICQ to determine the amount of rewards
+		// The rewards will get to the delegation account in the callback
+		err = utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
+			return k.SubmitWithdrawalBalanceICQ(ctx, hostZone)
```

Comment:
This is just a more accurate rename of the function which does the same work, but what about the "second half" of the logic here found in the callback (in keeper/icqcallbacks_withdrawl_balance.go)  Should we also wrap that in the same type of rollback cache?

Reply:
the callbacks are processed as tx's from the relayer so they already atomically succeed/fail - no need for cache wrapper!